src/engine/cachingreader/cachingreaderworker.cpp

```diff
@@ -114,27 +118,31 @@ void CachingReaderWorker::run() {
     }
 }
 
-void CachingReaderWorker::loadTrack(const TrackPointer& pTrack) {
+void CachingReaderWorker::cleanUpOldTrack() {
+    m_pAudioSource.reset(); // Close open file handles of the old track.
+
     // Discard all pending read requests
     CachingReaderChunkReadRequest request;
```

Comment:
The while loop could be moved into a separate function: `discardAllPendingRequests()` and the comment then becomes obsolete.